From 602f2f7199cac1226f423685e6df90d77fa3eb24 Mon Sep 17 00:00:00 2001
From: Sourabh Banerjee <sbanerje@codeaurora.org>
Date: Fri, 29 Jun 2018 16:54:28 +0530
Subject: [PATCH] evdev: skip input devices that are pure accelerometer

In case an input device has tags along with ACCELEROMETER
attempt to create an evdev dispatcher.
---
 src/evdev.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/evdev.c b/src/evdev.c
index a680414..1902b86 100644
--- a/src/evdev.c
+++ b/src/evdev.c
@@ -2572,6 +2572,12 @@ evdev_configure_device(struct evdev_device *device)
 	enum evdev_device_udev_tags udev_tags;
 	unsigned int tablet_tags;
 	struct evdev_dispatch *dispatch;
+	#define EVDEV_UDEV_TAG_NON_ACCELEROMETER \
+		  (EVDEV_UDEV_TAG_KEYBOARD | EVDEV_UDEV_TAG_MOUSE | \
+		   EVDEV_UDEV_TAG_TOUCHPAD | EVDEV_UDEV_TAG_TOUCHSCREEN | \
+		   EVDEV_UDEV_TAG_TABLET | EVDEV_UDEV_TAG_JOYSTICK | \
+		   EVDEV_UDEV_TAG_TABLET_PAD | EVDEV_UDEV_TAG_POINTINGSTICK | \
+		   EVDEV_UDEV_TAG_TRACKBALL | EVDEV_UDEV_TAG_SWITCH)
 
 	udev_tags = evdev_device_get_udev_tags(device, device->udev_device);
 
@@ -2596,7 +2602,8 @@ evdev_configure_device(struct evdev_device *device)
 		 udev_tags & EVDEV_UDEV_TAG_TRACKBALL ? " Trackball" : "",
 		 udev_tags & EVDEV_UDEV_TAG_SWITCH ? " Switch" : "");
 
-	if (udev_tags & EVDEV_UDEV_TAG_ACCELEROMETER) {
+	if ((udev_tags & EVDEV_UDEV_TAG_ACCELEROMETER)
+	    && (!(udev_tags & EVDEV_UDEV_TAG_NON_ACCELEROMETER))) {
 		evdev_log_info(device,
 			 "device is an accelerometer, ignoring\n");
 		return NULL;
-- 
1.9.1

